doc/Requirements.md

```diff
@@ -0,0 +1,526 @@
+(Why: For easy access from C code.)
+
+Supported integer size SHOULD be at least 8, 16 and 32 bits.
+(Why: The reference implementation supports it.)
+
+PF MAY support constraining the parameter minimum and maximum value.
+(Why: To catch user out of valid range errors when changing the parameter value.)
+
+##### String
+PF SHOULD support array of characters.
+(Why: Everything that a computer can store fits in an array of characters.
+      It can be used as a fallback type if no other matches the parameter.)
+
+The array maximum size MAY be invariant (immutable).
+(Note: This is what the reference implementation does.)
+
+The API to access the string value SHOULD support null terminated character array.
+As it is commonly done in C.
+(Why: For easy access from C code.)
+
+##### Fix point parameter
+PF SHOULD support fix point parameters. Ie integers divided by a fixed power of two.
+(Why: The reference implementation supports it.)
+
+The API to access the values SHOULD respect the Qm.n and UQm.n standards.
+(Why: It is the main standard for fix point parameters.)
+
+PF SHOULD support at least m + n ∈ [0,32]
+(Why: The reference implementation supports it.)
+
+PF MAY support constraining the parameter minimum and maximum value.
+(Why: To catch user out of valid range errors when changing the parameter value.)
+
+##### Floating point
+PF SHOULD support floating point parameters .
+(Why: The reference implementation supports it.)
+
+The API to access the values SHOULD respect the IEEE 754 standard.
+
+PF SHOULD support at least 32 and 64 bit size floats.
+(Why: The reference implementation supports it.)
+
+##### Bit field following the C abi.
+PF SHOULD support 1 or more bit sized integers.
+(Why: The reference implementation supports it.)
+
+Such parameters called are regrouped in a so called bit parameter block.
+The API to access a bit parameter block SHOULD give access to a packed bit field.
+(Why: The reference implementation supports it.)
+
+### Parameter adaptation
+
+#### Philosophy
+Parameters expose by hardware sometimes need to transform a parameter
+value. For example an hardware integer parameter could have a range 64-128 but
+it might be necessary for upper layer to access in a range 0-100.
+
+This transformation (called parameter adaptation in the rest of the document)
+could be done by the syncer. Nevertheless syncer are supposed to contain only
+business logic and should not be impacted by upper layer needs.
+
+#### Definition
+Parameter adaptation is a bijective pure function converting a parameter
+value between the syncer and other parameter reader/writer (including the inference engine).
+
+#### Requirements
+The following parameter adaptation SHOULD be supported
+ - Affine adaptation:    `affAd(value) = slope * value + offset` where slope and offset and constant coefficient
+(Why: The reference application supports it.)
+ - Logarithm adaptation: `logAd(value) = log(value)` where log() is the binary logarithm
```

Comment:
I don't think it is necessary to mention non-requirements (there are an infinity of those). If we ever need exponential adaptation, we'll add it.